doc/project/detailed_contribution_guide/_index.md

```diff
@@ -0,0 +1,340 @@
+We use separate style guides for different languages:
+* [SystemVerilog](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md)
+* [C/C++]({{< relref "c_cpp_coding_style.md" >}})
+* [Python]({{< relref "python_coding_style.md" >}})
+* [Markdown]({{< relref "markdown_usage_style.md" >}})
+* [Hjson]({{< relref "hjson_usage_style.md" >}})
+* [RISC-V Assembly]({{< relref "asm_coding_style.md" >}})
+
+If unsure about the style, be consistent with the existing code and do your best to match its style.
+
+For C/C++ code, we recommend to run "git-clang-format" on your changes.
+
+## What should commit messages look like?
+
+There are many [useful guides](https://chris.beams.io/posts/git-commit/) available on how to write commit messages and why.
+
+The following list contains a set of simple guidelines that we consider most useful.
+* A commit message consists of a subject line followed by an optional body.
+* Keep the subject line short. Ideally under 50 characters.
+* Capitalize the subject line.
+* Use the imperative mood in the subject line.
+* Use the present tense ("Add feature" not "Added feature").
+* When changes are restricted to a specific area, you should add a tag to the start of the subject line in square brackets, e.g. "\[uart/rtl\] Rework Tx FSM". See also [subject tags](#subject-tags).
+* Do not end the subject line with a period.
+* Separate subject from body with a blank line.
+* Wrap the body at 72 characters.
+* Use the body to explain what and why (rather than how).
+
+To indicate acceptance of our Contributor License Agreement (CLA), commits should also include a Signed-off-by line (which can be generated by `git commit -s`).
+See [CONTRIBUTING.md](https://github.com/lowRISC/opentitan/blob/master/CONTRIBUTING.md) for more information.
+
+## Subject tags
+
+If code changes, issues or pull requests (PRs) are restricted to a specific area, we recommend adding a tag in square brackets, *e.g.*, "\[uart/rtl\]", to the subject line of your commit message, issue or PR.
+Such tags help to identify areas affected by changes.
+Tags are particularly helpful when dealing with pull requests (PRs) as they can guide team members for code review and simplify later retrieval of specific PRs through the GitHub web interface.
+
+You can add multiple tags separated by a comma, e.g., "\[top, dv\]" to indicate that multiple separated areas are affected.
+In addition, you can add further levels of hierarchy to make the indication more precise such as "\[uart/rtl\]" instead of just "\[uart\]", "\[sw/device\]" instead of "\[sw\]".
+
+## What about TODO comments?
+
+We track non-trivial pieces of work via GitHub issues rather than TODO comments in the code.
+We do not use TODO(name) comments as used in some code bases - if it is important enough to assign someone, it is worth tracking via an issue.
+
+If you as a code reviewer come across TODOs, you should encourage the author to create a GitHub issue to track the work item.
```

Comment:
Can we recommend a format for TODO comments linking to GitHub issues, for example:

```
// TODO(lowrisc/<project_name>#<issue_number>): <brief description>
```

Reply:
Philipp reminded me there was some text on this [in the C/C++ style guide](https://docs.opentitan.org/doc/rm/c_cpp_coding_style/#todo-comments). `// TODO: This algorithm should be rewritten to be more efficient.
// (Bug lowrisc/reponame#27)`. How about I import that example here, and if anyone feels that a slightly different format would be better, we cover in a follow-up issue?

Reply:
For my part, I have been using the format suggested by @moidx.

Reply:
There was a fair amount of discussion within the project to come to the guidance that is now part of the C/C++ style guide. Even though we can certainly open that can again, I would suggest not doing so as part of this contributor guide discussion.

Reply:
I prefer a format that is then later easier to parse by machines, so that we can implement automation around it. That is part of the reason I prefer the issue to be inline with the TODO marker.  I am not really sure what are the arguments in favor for the format used in the style guide, but would be happy to revisit them if needed. 

I am supportive of which ever format works best with automation.

Reply:
Tooling should probably be written to just find all references to lowrisc/.+/#[0-9]+. Unless I've completely misremembered I think most discussion about TODOs in comments were around mentioning the assignee or not (we decided not) rather than the precise syntax of referring to issues.

As Miguel's proposed format seems nice and easy to read, I've added that as an example. Turning it into a requirement to use this precise format and adjusting any in-tree TODO comments is best tracked as a separate issue I think.

Is everyone happy with that resolution?